### PR TITLE
Added Teacher and Admin dashboards and Login

### LINKED ISSUE #52: Create Teacher Homepage
### Description
Create the homepage that a teacher gets redirected to once signing in.  This will use the username of the current user provided in #34.  This page will display the teacher's information (user name) as well as display the classes they are currently teaching to.  Once a class is selected, a teacher should be able to see a list of reviews completed by students for that class.  These can be presented modally or in dropdown fashion.

### PR DESCRIPTION
**Pull request by ciarapowell**
Fixes #52 , #53, #118 .

Changes made in this pull request:
Used userTypes to configure review page. 
Added Dashboard information to all users.

### CODE REVIEW

File: src/app/admin-dashboard/admin-dashboard.component.ts [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
 
   ngOnInit() {
+    this.currentAdministrator = this.service.getAdministrator(this.administratorID);
+    this.courses = this.service.getCoursesByAdministrator(this.administratorID);
   }
-
+  goToSubmitReviews(userId: string) {
+    this.router.navigate(['/admin-dashboard/review/', userId]);
 }
+}
```
Comment (ltcarbonell) [inline src/app/admin-dashboard/admin-dashboard.component.ts]: this is alot of extra newlines, they can probably go away

File: src/app/review/review.component.ts [hunk 2]
```diff
@@ -36,16 +39,33 @@ export class ReviewComponent implements OnInit {
...
     ) {
         this.route.params.subscribe(params => this.userID = params._id);
+        this.route.url.subscribe(url => this.userURL);
     }
 
     selectedValue: string;
 
     ngOnInit(): void {
+        if(this.router.url.indexOf('student') > 0){
+                     this.userType =1;
```
Comment (ltcarbonell) [inline src/app/review/review.component.ts]: formatting here needs to be adjusted to match the rest of this block

File: src/app/review/review.component.ts [hunk 3]
```diff
@@ -70,13 +90,63 @@ export class ReviewComponent implements OnInit {
                 this.coursesReviewed = Promise.resolve(coursesReviewed);
             });
             this.questions = this.reviewService.getQuestions();
-        } else { // Other user types
+        } else if(this.userType == 2){
+            this.user = this.teacherService.getTeacher(this.userID);
+            Promise.all([
+                this.courses = this.teacherService.getCoursesByTeacher(this.userID),
+                this.reviews = this.teacherService.getReviewsByTeacher(this.userID),
```
Comment (ltcarbonell) [inline src/app/review/review.component.ts]: we only need `this.reviews` here, `this.courses` can be removed.

File: src/app/review/review.component.ts [hunk 4]
```diff
@@ -70,13 +90,63 @@ export class ReviewComponent implements OnInit {
...
+                        coursesReviewed.push({
+                            course: course,
+                            review: reviews.find(reviewedCourse => reviewedCourse.course_code === course.course_code)
+                        });
+                    }
+                });
+                this.coursesToBeReviewed = Promise.resolve(coursesToBeReviewed);
+                this.coursesReviewed = Promise.resolve(coursesReviewed);
+            });
+            this.questions = this.reviewService.getQuestions();
```
Comment (ltcarbonell) [inline src/app/review/review.component.ts]: This is not needed

File: src/app/review/review.component.ts [hunk 5]
```diff
@@ -70,13 +90,63 @@ export class ReviewComponent implements OnInit {
...
+
+        } else if(this.userType == 3){
```
Comment (ltcarbonell) [inline src/app/review/review.component.ts]: Similar (if not the same) changes need to happen here as above with the teacher's section

File: src/app/review/review.component.ts [hunk 6]
```diff
@@ -70,13 +90,63 @@ export class ReviewComponent implements OnInit {
                 this.coursesReviewed = Promise.resolve(coursesReviewed);
             });
             this.questions = this.reviewService.getQuestions();
-        } else { // Other user types
+        } else if(this.userType == 2){
+            this.user = this.teacherService.getTeacher(this.userID);
+            Promise.all([
+                this.courses = this.teacherService.getCoursesByTeacher(this.userID),
+                this.reviews = this.teacherService.getReviewsByTeacher(this.userID),
+            ]).then(data => {
```
Comment (ltcarbonell) [inline src/app/review/review.component.ts]: This logic needs here to be slightly modified.  We don't need a list of courses that needs to be reviewed.  Instead we just need to display the reviews that have already taken place.

for example:

```
const coursesReviewed = [];
reviews.forEach(review => {
    coursesReviewed.push(review);
});
```

The structure of the return will probably need to be modified to account for the expected values in the html file.

File: src/app/review/review.component.ts (hunk 3 above)
Comment (ciarapowell) [reply src/app/review/review.component.ts]: I would prefer to keep the course information with the review. The review only has the course id correct ?

Comment (ltcarbonell) [reply src/app/review/review.component.ts]: No it has some of the course information as well.

Here is an example:
``` json
{
    "course_name": "Software Engineering",
    "course_code": "CEN5035",
    "review_content": [
    {
        "answer": "Yeah they are great",
        "question": "Are they a good teacher",
        "label": "",
        "required": false,
        "order": 1,
        "controlType": "textbox",
        "type": ""
    },
    {
        "answer": "Nope",
        "question": "Any comments",
        "label": "",
        "required": false,
        "order": 1,
        "controlType": "textbox",
        "type": ""
    }
]
},
```

**Pull request merged**